compiler/rustc_hir_typeck/src/fn_ctxt/_impl.rs

```diff
@@ -1423,7 +1423,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             let impl_ty = self.normalize(span, tcx.type_of(impl_def_id).instantiate(tcx, args));
             let self_ty = self.normalize(span, self_ty);
             match self.at(&self.misc(span), self.param_env).eq(
-                DefineOpaqueTypes::No,
+                DefineOpaqueTypes::Yes,
```

Comment:
I don't know how we can create a test for this... This needs to be a FIXME and we need to find out how to make a test.